torch/csrc/jit/python/init.cpp

```diff
@@ -1655,7 +1682,12 @@ void initJITBindings(PyObject* module) {
   // because otherwise prim::Print() instruction won't work for JIT modules.
   auto atexit = py::module_::import("atexit");
   atexit.attr("register")(
-      py::cpp_function([]() { setPrintHandler(getDefaultPrintHandler()); }));
+      py::cpp_function([]() { 
+        setPrintHandler(getDefaultPrintHandler());
```

Comment:
@ezyang we will need to clear the table when we are unloading python modules since otherwise it will be too late: all python classes will be gone but the table may still contain `PythonSymbolicIntNode` pointing to python objects.

Reply:
Use PyInterpreter decref for C++ to Python references (SafePyObject will do this automatically for you), this will ensure that we just leak the PyObject if CPython is already dead by the time of destruction.